javatests/arcs/android/storage/service/StorageServiceEndToEndTest.kt

```diff
@@ -94,17 +103,63 @@ class StorageServiceEndToEndTest {
         DriverFactory.clearRegistrations()
     }
 
+    @Test
+    fun writeThenRead_nullInlines_inCollection_onDatabase() = runBlocking {
+        val handle = createCollectionHandle(databaseKey)
+        val entity = emptyEntityForTest()
+
+        handle.dispatchStore(entity)
+
+        val handle2 = createCollectionHandle(databaseKey)
+        assertThat(handle2.dispatchFetchAll()).containsExactly(entity)
+        Unit
+    }
+
     @Test
     fun writeThenRead_inlineData_inCollection_onDatabase() = runBlocking {
         val handle = createCollectionHandle(databaseKey)
         val entity = entityForTest()
 
         handle.dispatchStore(entity)
 
+        databaseManager.runGarbageCollection()
+        databaseManager.runGarbageCollection()
+
         val handle2 = createCollectionHandle(databaseKey)
-        val data = handle2.fetchAll()
-        assertThat(data.size).isEqualTo(1)
-        assertThat(data.toList()[0]).isEqualTo(entity)
+        assertThat(handle2.dispatchFetchAll()).containsExactly(entity)
+        Unit
+    }
+
+    @Test
+    fun writeThenRead_inlineData_inCollection_onDatabase_withExpiry() = runBlocking {
+        val handle = createCollectionHandle(databaseKey, Ttl.Hours(1))
+        val entity = entityForTest()
+        val entity2 = entityForTest()
+
+        time.millis = System.currentTimeMillis()
+        handle.dispatchStore(entity)
+        time.millis = 1L
+        handle.dispatchStore(entity2)
+
```

Comment:
to reproduce the bug we had, you would need to call databaseManager.removeExpiredEntities() here